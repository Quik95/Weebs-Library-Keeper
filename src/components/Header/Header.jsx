import React from 'react';

// MUI
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import InputBase from '@material-ui/core/InputBase';
import SearchIcon from '@material-ui/icons/Search';

// styles
import useStyles from './Header.styles';

// custom components
import SearchResults from './SearchResults/SearchResultsContainer';

export default function SearchAppBar() {
  const classes = useStyles();

  return (
    <header className={classes.searchResultsFix} data-testid="Header">
      <AppBar position="static">
        <Toolbar>
          <Typography className={classes.title} variant="h6" noWrap>
            Weeb's Library Keeper
          </Typography>
          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <InputBase
              placeholder="Searchâ€¦"
              classes={{
                root: classes.inputRoot,
                input: classes.inputInput,
              }}
              inputProps={{ 'aria-label': 'search' }}
            />
            <SearchResults />
          </div>
        </Toolbar>
      </AppBar>
    </header>
  );
}
