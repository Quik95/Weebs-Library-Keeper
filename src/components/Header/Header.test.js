import React from 'react';
import { render } from '@testing-library/react';

import Header from './Header';

describe('<Header/>', () => {
  const { getByTestId } = render(<Header />);

  it('Renders correctly', () => {
    const wrapper = getByTestId('Header');
    expect(wrapper).toMatchSnapshot();
  });
});
