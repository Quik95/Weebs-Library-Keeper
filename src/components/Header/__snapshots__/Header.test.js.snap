// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Renders correctly 1`] = `
<header
  class="makeStyles-searchResultsFix-1"
  data-testid="Header"
>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h6 MuiTypography-noWrap"
      >
        Weeb's Library Keeper
      </h6>
      <div
        class="makeStyles-search-4"
      >
        <div
          class="makeStyles-searchIcon-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          class="MuiInputBase-root makeStyles-inputRoot-6"
        >
          <input
            aria-label="search"
            class="MuiInputBase-input makeStyles-inputInput-7"
            placeholder="Searchâ€¦"
            type="text"
            value=""
          />
        </div>
        <ul
          class="MuiList-root makeStyles-listContainer-108 MuiList-padding"
          data-testid="SearchContainer"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
            data-testid="ListItem"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13.127 14.56l1.43-1.43 6.44 6.443L19.57 21zm4.293-5.73l2.86-2.86c-3.95-3.95-10.35-3.96-14.3-.02 3.93-1.3 8.31-.25 11.44 2.88zM5.95 5.98c-3.94 3.95-3.93 10.35.02 14.3l2.86-2.86C5.7 14.29 4.65 9.91 5.95 5.98zm.02-.02l-.01.01c-.38 3.01 1.17 6.88 4.3 10.02l5.73-5.73c-3.13-3.13-7.01-4.68-10.02-4.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root makeStyles-listItemText-109 MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dr. Stone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                21/11/2019 18:50
              </p>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </header>
</header>
`;
